### ISSUE TITLE
32 bit pointer in scipy.ndimage.zoom()

### ISSUE THREAD

**Issue by tylersc2** [scipy/scipy#4638]
If I use "zoom" on an array which is more than 4GB, I get some artifacts which strongly suggest to me that a 32 bit pointer is being used to access the input array.  (I.e., the part of the image past the 4GB point ends up being a repeat of the 1st part of the image)

**Comment by rgommers** [comment]
@tylersc2 thanks for the bug report. Can you provide a self-contained example that demonstrates the issue?

**Comment by tylersc2** [comment]
Yes, I’d be happy to oblige (I’m impressed that you got on this so fast).
Please note that my python computer is NOT on any network so I am typing this in by hand (please forgive any typos).


import numpy as np
import matplotlib.pyplot as plt
import scipy.ndimage

n = 17000      # To show the problem the image needs to use more than 2^31 bytes of memory
#   (n**2) * (8 bytes/float64)  = 2312000000 > 2**31 bytes
a = np.linspace(0,1,n**2).reshape(n,n)  # n by n image with gradient from top to bottom
b = scipy.ndimage.interpolation.zoom(a,1)
plt.figure()
# 1st ~15785 rows show gradient, e.g., row 15780 is close to ‘1’
#   but last rows are all close to ‘0’
plt.imshow(b,interpolation=’nearest’)
plt.show()

**Comment by rgommers** [comment]
Don't know what's wrong with the code above, but here it is in copy-paste-able form:

    import numpy as np
    import matplotlib.pyplot as plt
    import scipy.ndimage 
    n = 17000 # To show the problem the image needs to use more than 2^31 bytes of memory 
    # (n**2) * (8 bytes/float64) = 2312000000 > 2**31 bytes 
    a = np.linspace(0,1,n**2).reshape(n,n) # n by n image with gradient from top to bottom 
    b = scipy.ndimage.interpolation.zoom(a,1) 
    plt.figure() 
    # 1st ~15785 rows show gradient, e.g., row 15780 is close to ‘1’
    # but last rows are all close to ‘0’
    plt.imshow(b,interpolation='nearest')
    plt.show()

**Comment by rgommers** [comment]
Did a quick check: ``NI_ZoomShift`` and the iterators in ``ni_support.c`` seem to be careful to use ``npy_intp`` everywhere, except for a few places like loop indices over number of array dimensions. 

No time to do more than a superficial check.
